Add new breaking changes on Java/Android (#3695)
getsentry/sentry-docs: src/platforms/android/migration.mdx

`SentryOptions#getCacheDirSize` has been deprecated in favor of `SentryOptions#getMaxCacheItems`.

> `InvalidDsnException` has been removed in favor of `IllegalArgumentException`.

Reviewer: ```suggestion
`InvalidDsnException` has been removed. It is replaced by `IllegalArgumentException`.
```

> `EventProcessor` interface has a new `default` method and this could break the instantiation when using trailing lambdas.

Reviewer: ```suggestion
`EventProcessor` interface has a new `default` method which could break the instantiation when using trailing lambdas.
```
